core/src/main/java/org/infinispan/util/IntSet.java

```diff
@@ -0,0 +1,196 @@
+package org.infinispan.util;
+
+import java.util.BitSet;
+import java.util.Collection;
+import java.util.PrimitiveIterator;
+import java.util.Set;
+
+/**
+ * Represent a set of integers (e.g. segments) as a {@code BitSet}.
+ *
+ * @author Dan Berindei
+ * @since 9.0
+ */
+public class IntSet implements Set<Integer> {
```

Comment:
If that means that you'll remove `IntSet` in the next JIRA, use `BitSet` right away. Otherwise the class will stay rotting in the codebase.